atomicapp/nulecule/main.py

```diff
@@ -185,14 +158,55 @@ def init(app_name, destination=None, app_version='1.0',
                     os.remove(templ_path)
 
             # Copy rendered templates to destination directory
-            distutils.dir_util.copy_tree(tmpdir, destination, True)
+            copy_tree(tmpdir, destination, True)
         finally:
             # Remove temporary working directory
-            distutils.dir_util.remove_tree(tmpdir)
+            remove_tree(tmpdir)
         return destination
 
-    def unpack(self, update=False,
-               dryrun=False, nodeps=False, config=None):
+    def digraph(self, update=False, dryrun=False, nodeps=False, config=None,
+                **kwargs):
+        """
+        Args:
+            update (bool): Update existing Nulecule application in
+                           app_path, if True
+            dryrun (bool): Do not make any change to the host system
+            nodeps (bool): Do not unpack external dependencies
+            config (dict): Config data, if any, to use for unpacking
+            kwargs (dict): Extra keyword arguments
+
+        Returns:
+            networkx.DiGraph of Nulecule objects
+        """
+        digraph = DiGraph()
+
+        (progenitor,
+         internals,
+         externals) = self.unpack(config=config, dryrun=dryrun, nodeps=nodeps,
+                                  update=update, **kwargs)
+
+        digraph.add_node(progenitor)
+
+        for internal in internals:
+            digraph.add_edge(progenitor, internal)
+
+        for external in externals:
+            app_spec = Utils.getSourceImage(external)
+            name = external[NAME_KEY]
+            destination = "%s/%s/%s" % (self.app_path, EXTERNAL_APP_DIR, name)
+            if app_spec is not None:
+                kwargs.update({'namespace': name})
+                kwargs.update(self._process_app_spec(app_spec, destination))
+                subgraph = self.digraph(**kwargs)
+                if isinstance(subgraph, DiGraph) and subgraph.order() > 0:
```

Comment:
Yeah, that's an artifact from a previous rendition of the code. I will clip that.